code/modules/overmap/ships/computers/sensors.dm

```diff
@@ -4,12 +4,18 @@
 	icon_screen = "teleport"
 	light_color = "#77fff8"
 	extra_view = 4
-	var/obj/machinery/shipsensors/sensors
+	var/weakref/sensor_ref
 	var/list/last_scan
 	var/working_sound = 'sound/machines/sensors/dradis.ogg'
 	var/datum/sound_token/sound_token
 	var/sound_id
 
+/obj/machinery/computer/ship/sensors/proc/get_sensors()
+	var/obj/machinery/shipsensors/sensors = sensor_ref?.resolve()
+	if(!istype(sensors) || QDELETED(sensors))
```

Comment:
Merely a comment not a request for change: A weakref should never resolve a deleted instance, as long as qdel/Destroy works as intended